test/alternator/test_system_tables.py

```diff
@@ -5,22 +5,29 @@
 # Tests for accessing alternator-only system tables (from Scylla).
 
 import pytest
+import requests
+
 from botocore.exceptions import ClientError
 from boto3.dynamodb.conditions import Key
 
 internal_prefix = '.scylla.alternator.'
 
 # Test that fetching key columns from system tables works
-def test_fetch_from_system_tables(scylla_only, dynamodb):
+def test_fetch_from_system_tables(scylla_only, dynamodb, rest_api):
     client = dynamodb.meta.client
     tables_response = client.scan(TableName=internal_prefix+'system_schema.tables',
             AttributesToGet=['keyspace_name','table_name'])
 
+    # #13332 - don't rely on "system" prefix to decide what is user keyspace or not.
+    resp_user = requests.get(f'{rest_api}/storage_service/keyspaces?type=user', timeout=1)
+    resp_user.raise_for_status()
+    keyspaces_user = resp_user.json()
+
     for item in tables_response['Items']:
         ks_name = item['keyspace_name']
         table_name = item['table_name']
 
-        if not 'system' in ks_name:
+        if ks_name in keyspaces_user:
```

Comment:
There is a risk here that if the query returns (because it is buggy) all keyspaces, not just user keyspaces, it just skips doing the real test it wanted to do.
Maybe if you don't "continue", you can set "any_system_keyspace=true" (or similar) and at the end of the test assert any_system_keyspace.